### PR TITLE
Add tox for running tests with different Python version

### LINKED ISSUE #2: Setup Github Actions to run tests and linters
Setup Github Actions to run tests and linters

### LINKED ISSUE #4: Use tox to run tests using different Python versions
Use tox to run tests using different Python versions.

### PR DESCRIPTION
**Pull request by ollipa**
Add tox for running tests with different Python version.

Closes #2
Closes #4

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Types of changes:
 
 ### Added
 
-- Add Python 3.6, 3.7, 3.8, 3.9, 3.10 support.
+- Add Python 3.6, 3.7, 3.8, 3.9 support.
```
Comment (christophe-riolo) [inline CHANGELOG.md:19]: Is tox limiting the 3.10 yet or did we find bugs ?

File: Makefile [hunk 2]
```diff
@@ -10,11 +10,22 @@ all: lint test
...
+	pytest tests/flexmock_pytest_test.py
+
+.PHONY: unittest
+unittest:
 	@printf '\n\n*****************\n'
-	@printf '$(color)Running tests$(off)\n'
+	@printf '$(color)Running unittest$(off)\n'
 	@printf '*****************\n'
-	pytest
+	PYTHONPATH=$(shell pwd) python tests/flexmock_test.py
```
Comment (christophe-riolo) [inline Makefile:28]: Why do you need to set the pythonpath ?

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[tox]
+isolated_build = True
+envlist = pypy3.6, pypy3.7, py36, py37, py38, py39
```
Comment (christophe-riolo) [inline tox.ini:3]: We could update the metadata in pyproject to say precisely which python versions we support

File: CHANGELOG.md (hunk 1 above)
Comment (ollipa) [reply CHANGELOG.md:19]: Python 3.10 works with Tox. If I have no trouble setting it up Github Actions, I'll enable tests for it too.

File: Makefile (hunk 2 above)
Comment (ollipa) [reply Makefile:28]: I didn't really try to debug it further but without setting the current folder to PYTHONPATH I get this error:
```
python tests/flexmock_test.py
Traceback (most recent call last):
  File "/Projects/python/flexmock/tests/flexmock_test.py", line 28, in <module>
    from tests import some_module
ModuleNotFoundError: No module named 'tests'
make: *** [Makefile:28: unittest] Error 1
```

File: CHANGELOG.md (hunk 1 above)
Comment (ollipa) [reply CHANGELOG.md]: I added Python 3.10 back. CI pipeline now runs tests with the latest development version.

File: Makefile (hunk 2 above)
Comment (ollipa) [reply Makefile]: Solved this by running unittest with `python -m unittest`

**Review by christophe-riolo — APPROVED** [review]

**Pull request merged**